### ISSUE TITLE
MultiGPU Inefficiency

### ISSUE THREAD

**Issue by JohnnyRisk** [junyanz/pytorch-CycleGAN-and-pix2pix#539]
This implementation is great, thank you very much. I have a question though regarding the efficiency of training on multipleGPU's. The problem I am having is that when training on 8 GPU's it appears that GPU 0 is doing SIGNIFICANTLY more work than any other gpu. For example, I am using 8 gpus with 16gb's of memory each and the first gpu is fully utilized but the others only use 11gb's. Any thoughts about increasing the efficiency would be appreciated. An example of the utilization can be found below.

![gpu_utilization](https://user-images.githubusercontent.com/18605542/53040897-d5704700-343f-11e9-9f4f-97610ac67a78.png)

**Comment by taesungp** [comment]
Usually GPU 0 does consume more memory than the others because it needs to do the work of combining the gradients from the other GPUs. I haven't thought deeply about how to remedy this. Maybe you can modify DataParallel so that GPU0 is only responsible for combining the gradients, without actually doing backprop? It would be appreciated if you can explore and let us know.
